history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskanaHistoryEngineImpl.java

```diff
@@ -107,7 +112,27 @@ protected SqlSessionManager createSqlSessionManager() {
             DEFAULT, this.transactionFactory, taskanaEngineConfiguration.getDatasource());
     Configuration configuration = new Configuration(environment);
 
+    // set databaseId
+    String databaseProductName;
+    try (Connection con = taskanaEngineConfiguration.getDatasource().getConnection()) {
+      databaseProductName = DB.getDatabaseProductName(con);
+      String databaseProductId = DB.getDatabaseProductId(databaseProductName);
+      configuration.setDatabaseId(databaseProductId);
+
+    } catch (SQLException e) {
+      throw new SystemException(
+          "Method createSqlSessionManager() could not open a connection "
+              + "to the database. No databaseId has been set.",
+          e.getCause());
```

Comment:
why do you pass the cause of the SQLException? We lose some information this way :) 
Personally I also prefer not to use a method name hardcoded in a string. This screams like technical debt if this gets refactored/renamed some day.
How about "Could not open a connection to set the databaseId"?

Reply:
my bad. changed.